### ISSUE TITLE
Server crash on returning to the Overworld when no .json file is created

### ISSUE THREAD

**Issue by Dialupman** [lumien231/Perfect-Spawn#18]
Tested from a fresh install of Minecraft with only Perfect Spawn installed. I did not create the .json file initially since I had no idea what I was doing at the time, so my modpack repeatedly crashed every time I went back to the overworld from any dimension.

Since then, I learned how to use the mod and it is working as I need it to, but it left a bad taste in my mouth. As easy as it is to create and configure the .json file, I spent at least 4 hours trying to figure out what was causing my modpack to crash as I had added roughly 150 mods to my pack initially. While not necessary, adding some kind of config file to create a blank .json file that essentially recreates vanilla spawn conditions in newly created worlds would be quite user friendly and would prevent this crash from happening.

Steps to reproduce:
1. Install Perfect Spawn, but do not create the .json file
2. Travel to a dimension other than the Overworld.
3. Attempt to return to the Overworld.

---- Minecraft Crash Report ----

WARNING: coremods are present:
  LoadingPlugin (PerfectSpawn-MC1.12.2-2.0.jar)
Contact their authors BEFORE contacting forge

// My bad.

Time: 3/8/18 5:48 PM
Description: Exception ticking world

java.lang.NoSuchFieldError: world
	at net.minecraft.world.gen.ChunkProviderServer.func_73156_b(ChunkProviderServer.java:276)
	at net.minecraft.world.WorldServer.func_72835_b(WorldServer.java:207)
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:754)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:666)
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:185)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:524)
	at java.lang.Thread.run(Thread.java:745)

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Server thread
Stacktrace:
	at net.minecraft.world.gen.ChunkProviderServer.func_73156_b(ChunkProviderServer.java:276)
	at net.minecraft.world.WorldServer.func_72835_b(WorldServer.java:207)

-- Affected level --
Details:
	Level name: New World
	All players: 0 total; []
	Chunk stats: ServerChunkCache: 0 Drop: 1
	Level seed: -2855655428947090118
	Level generator: ID 00 - default, ver 1. Features enabled: false
	Level generator options: 
	Level spawn location: World: (0,0,0), Chunk: (at 0,0,0 in 0,0; contains blocks 0,0,0 to 15,255,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,0,0 to 511,255,511)
	Level time: 0 game time, 0 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: ~~ERROR~~ NullPointerException: null
Stacktrace:
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:754)
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:666)
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:185)
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:524)
	at java.lang.Thread.run(Thread.java:745)

-- System Details --
Details:
	Minecraft Version: 1.12.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 1.8.0_25, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 211228008 bytes (201 MB) / 1608515584 bytes (1534 MB) up to 4772593664 bytes (4551 MB)
	JVM Flags: 4 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx5120m -Xms256m -XX:PermSize=256m
	IntCache: cache: 0, tcache: 0, allocated: 12, tallocated: 94
	FML: MCP 9.42 Powered by Forge 14.23.2.2624 5 mods loaded, 5 mods active
	States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored

	| State     | ID           | Version      | Source                        | Signature                                |
	|:--------- |:------------ |:------------ |:----------------------------- |:---------------------------------------- |
	| UCHIJAAAA | minecraft    | 1.12.2       | minecraft.jar                 | None                                     |
	| UCHIJAAAA | mcp          | 9.42         | minecraft.jar                 | None                                     |
	| UCHIJAAAA | FML          | 8.0.99.99    | forge-1.12.2-14.23.2.2624.jar | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| UCHIJAAAA | forge        | 14.23.2.2624 | forge-1.12.2-14.23.2.2624.jar | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| UCHIJAAAA | perfectspawn | 2.0          | PerfectSpawn-MC1.12.2-2.0.jar | None                                     |

	Loaded coremods (and transformers): 
LoadingPlugin (PerfectSpawn-MC1.12.2-2.0.jar)
  lumien.perfectspawn.asm.ClassTransformer
	GL info: ~~ERROR~~ RuntimeException: No OpenGL context found in the current thread.
	Profiler Position: N/A (disabled)
	Player Count: 1 / 8; [EntityPlayerMP['Dialupman'/1213, l='New World', x=137.10, y=68.00, z=244.67]]
	Type: Integrated Server (map_client.txt)
	Is Modded: Definitely; Client brand changed to 'fml,forge'

**Comment by Dialupman** [comment]
So... I went back to making my pack and properly set up the .json file, but I am still getting repeated crashes.

If I set the file to spawn in the nether, I can freely teleport between the overworld and the nether, but if I try to teleport to another dimension, such as the beneath, the pack will crash upon return to the overworld. I also did more testing with the mod by itself and couldn't get it working again.

This is my perfectspawn.json and I have it located in the main directory of the save file. 

{
"general": {
"initialSpawnDimension": 0
},
"rules": [
]
}
